### DISCUSSION TITLE
[QUESTION]

### DISCUSSION THREAD

**Discussion by DrMartiner** [python-telegram-bot/python-telegram-bot#3050] [Q&A]
### Issue I am facing

Hello, 

How can I start web pooling at 13.11 version? The WiKi page with `Application` class is not actually...

THe code below does't work...

Generate SSL cert:
```shell
openssl req -newkey rsa:2048 -sha256 -nodes -keyout private.key -x509 -days 3650 -out cert.pem -subj "/C=US/ST=New York/L=Brooklyn/O=Example Brooklyn Company/CN=example.com"
curl -F "url=https://example.com/" -F "certificate=@cert.pem" "https://api.telegram.org/bot<token>/setwebhook"
```

Run code:
```python
updater = Updater('....')
updater.dispatcher.add_handler(MessageHandler((Filters.command & Filters.regex("^/start$")), process_start))

updater.bot.set_webhook('....', certificate=open('cert.pem', 'r'))
updater.start_webhook(
    listen='0.0.0.0',
    port=8443,
    url_path='....',
    key='private.key',
    cert='cert.pem',
    webhook_url='https://example.com/'
)
updater.idle()
```

Best regards

### Operating System

Ubuntu

### Version of Python, python-telegram-bot & dependencies

```shell
python-telegram-bot = "^13.11"
```

**Comment by Poolitzer** [comment]
Wiki pages for v13 are at https://github.com/python-telegram-bot/v13.x-wiki/wiki. And the code you showed generated a webhook, not polling, to a server you don't own, so thats where the error is.

**Comment by DrMartiner** [comment]
> Wiki pages for v13 are at https://github.com/python-telegram-bot/v13.x-wiki/wiki. And the code you showed generated a webhook, not polling, to a server you don't own, so thats where the error is.

Hmmm... I did according the example:
```python
updater = Updater(TOKEN)
updater.start_webhook(listen="0.0.0.0", port=5000, url_path=TOKEN, webhook_url=f"https://example.com/{TOKEN}")
updater.idle()
```

but I saw nothing in Nginx's logs. What was wrong?

**Comment by DrMartiner** [comment]
Btw, I registered my WebHook by curl before started the code: `curl': curl -F "url=https://xample.com/" -F "certificate=@cert.pem" "https://api.telegram.org/bot<TOKEN>/setwebhook"`

**Reply by Bibo-Joshi** [reply]
`start_webhook` will automatically call `set_webhook`. If you need a custom certificate, make sure to pass it to `start_webhook`. Inspecting the output of `Bot.get_webhook_info` may also help debugging.

**Reply by DrMartiner** [reply]
> start_webhook will automatically call set_webhook. 

Yes, I know, but my code doesn't work that's why I try to find out the cause...

Is code in my question right?

**Reply by Poolitzer** [reply]
both code examples still mention example.com, put your own URL there...

**Reply by DrMartiner** [reply]
I set my real domain on by server, but I didn't see requests in Nginx's logs

**Reply by Bibo-Joshi** [reply]
> ```python
> updater.bot.set_webhook('....', certificate=open('cert.pem', 'r'))
> updater.start_webhook(
>     listen='0.0.0.0',
>     port=8443,
>     url_path='....',
>     key='private.key',
>     cert='cert.pem',
>     webhook_url='https://example.com/'
> )
> ```

here, `updater.bot.set_webhook('....', certificate=open('cert.pem', 'r'))` is superfluous and may lead to problems.
What's the output of `print(updater.bot.get_webhook_info)`?

**Reply by DrMartiner** [reply]
> > `updater.bot.get_webhook_info`

I did it. It will return WebHookInfo object. What fields should I print?

**Reply by Bibo-Joshi** [reply]
all of them. just run `print(updater.bot.get_webhook_info())` (sorry, I forgot the `()` in my previous post)
